src/days-2-4-adv/adv.py

```diff
@@ -49,3 +49,63 @@
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+print (f"""
+    {me.currentRoom.name}:
+
+    {me.currentRoom.description}
+    """)
+while True:
+    cmd = input("which way?-> ")
+    if cmd == "q":
+        print ("""
+    ~~~~ until tomorrow ~~~~
+        """)
+        break
+    if cmd == "n":
+        if hasattr(me.currentRoom.n_to, "name"):
+            me.currentRoom = me.currentRoom.n_to
+            print (f"""
+    {me.currentRoom.name}:
+
+    {me.currentRoom.description}
+    """)
+        else:
+            print ("""
+    Yo you cant go that way dog, duh, ya dingus
+            """)
+    elif cmd == "e":
+        if hasattr(me.currentRoom.e_to, "name"):
+            me.currentRoom = me.currentRoom.e_to
+            print (f"""
+    {me.currentRoom.name}:
+
+    {me.currentRoom.description}
+    """)
+        else:
+            print ("""
+    Yo you cant go that way dog, duh, ya dingus
+            """)
+    elif cmd == "s":
+        if hasattr(me.currentRoom.s_to, "name"):
+            me.currentRoom = me.currentRoom.s_to
+            print (f"""
+    {me.currentRoom.name}:
+
+    {me.currentRoom.description}
+    """)
+        else:
+            print ("""
+    Yo you cant go that way dog, duh, ya dingus
+            """)
+    elif cmd == "w":
+        if hasattr(me.currentRoom.w_to, "name"):
+            me.currentRoom = me.currentRoom.w_to
+            print (f"""
+    {me.currentRoom.name}:
+
+    {me.currentRoom.description}
+    """)
+        else:
+            print ("""
+    Yo you cant go that way dog, duh, ya dingus
+            """)
```

Comment:
After finishing MVP today or tomorrow, can you think of how you can make your game loop "DRY"er? (i.e. **D**_on't_ **R**_epeat_ **Y**_ourself_) For example, no matter what direction the player chooses, this will always happen:
```
            print (f"""
    {me.currentRoom.name}:
    {me.currentRoom.description}
    """)
```
Should that be within the `if...else` tree or outside of it? Have that in mind when you take another look at your code after you're finished.